### PR TITLE
[FIX-3573][DAO] potential horizontal unauthorized access

### PR DESCRIPTION
**Pull request by lenboo**
fix bug(#3573) potential horizontal unauthorized access

### CODE REVIEW

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapper.xml [hunk 1]
```diff
@@ -168,14 +168,14 @@
         select *
         from t_ds_process_instance
         where 1=1
+        and process_definition_id=#{processDefinitionId}
```
Comment (CalvinKirs) [inline dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapper.xml:171]: Hi, is it possible to delete this statement?

**Review by CalvinKirs — CHANGES_REQUESTED** [review]

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapper.xml [hunk 2]
```diff
@@ -168,14 +168,14 @@
         select *
         from t_ds_process_instance
         where 1=1
```
Comment (CalvinKirs) [inline dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapper.xml:170]: Hi, is it possible to delete this statement?

**Review by CalvinKirs — CHANGES_REQUESTED** [review]

**Review by lenboo — COMMENTED** [review]
thank you for reminding!
"1=1" is not necessary.

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapper.xml (hunk 2 above)
Comment (lenboo) [reply dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessInstanceMapper.xml:170]: thank you for reminding!
"1=1" is not necessary.

**Review by CalvinKirs — APPROVED** [review]
+1

**Review by lgcareer — APPROVED** [review]
+1

**Pull request merged**